src/ui/units/datasets/containers/Dataset/useHistoryActions.tsx

```diff
@@ -0,0 +1,96 @@
+import {goBack, goForward} from 'ui/store/actions/editHistory';
+import {selectCanGoBack, selectCanGoForward} from 'ui/store/selectors/editHistory';
+import Utils from 'ui/utils';
+
+import {DATASETS_EDIT_HISTORY_UNIT_ID} from '../../constants';
+
+import {ACTION_PANEL_ICON_SIZE} from './constants';
+
+const canGoBackSelector = (state: DatalensGlobalState) => {
+    const isValidationPending =
+        state.dataset.savingDataset.disabled && state.dataset.ui.isDatasetChanged;
+
+    return (
+        !isValidationPending &&
+        !state.dataset.validation.isLoading &&
+        selectCanGoBack(state, {unitId: DATASETS_EDIT_HISTORY_UNIT_ID})
+    );
+};
+
+const canGoForwardSelector = (state: DatalensGlobalState) => {
+    const isValidationPending =
+        state.dataset.savingDataset.disabled && state.dataset.ui.isDatasetChanged;
+
+    return (
+        !isValidationPending &&
+        !state.dataset.validation.isLoading &&
+        selectCanGoForward(state, {unitId: DATASETS_EDIT_HISTORY_UNIT_ID})
+    );
+};
+
+export function useHistoryActions() {
+    const dispatch = useDispatch();
+    const enabled = Utils.isEnabledFeature(Feature.EnableEditHistoryDatasets);
+    const canGoBack = useSelector(canGoBackSelector);
+    const canGoForward = useSelector(canGoForwardSelector);
+    const options: Options = React.useMemo(() => {
+        return {enabled, scopes: HOTKEYS_SCOPES.DATASETS};
+    }, [enabled]);
+
+    const handleGoBack = React.useCallback(() => {
+        if (canGoBack) {
+            dispatch(goBack({unitId: DATASETS_EDIT_HISTORY_UNIT_ID}));
+        }
+    }, [canGoBack, dispatch]);
+
+    const handleGoForward = React.useCallback(() => {
+        if (canGoForward) {
+            dispatch(goForward({unitId: DATASETS_EDIT_HISTORY_UNIT_ID}));
+        }
+    }, [canGoForward, dispatch]);
+
+    const items: AdditionalButtonTemplate[] = React.useMemo(() => {
+        if (!enabled) {
+            return [];
+        }
+
+        return [
+            {
+                key: 'undo',
+                action: handleGoBack,
+                icon: {data: ArrowUturnCcwLeft, size: ACTION_PANEL_ICON_SIZE},
+                view: 'flat',
+                disabled: !canGoBack,
+                title: i18n('component.action-panel.view', 'button_undo'),
+                hotkey: UNDO_HOTKEY.join('+'),
+            },
+            {
+                key: 'redo',
+                action: handleGoForward,
+                icon: {data: ArrowUturnCwRight, size: ACTION_PANEL_ICON_SIZE},
+                view: 'flat',
+                disabled: !canGoForward,
+                title: i18n('component.action-panel.view', 'button_redo'),
+                hotkey: REDO_HOTKEY.join('+'),
+            },
+        ];
+    }, [canGoBack, canGoForward, enabled, handleGoBack, handleGoForward]);
+    const actions = useAdditionalItems({items});
+
+    useBindHotkey({key: UNDO_HOTKEY, handler: handleGoBack, options});
```

Comment:
Now buttons are shown only if flag is true, but hotkeys are always bound. Hotkeys should be bound only if flag is true, too.